---
layout: project
title: 'Transcriptional control of immunity and inflammation'
image: /assets/img/projects/network1.png
screenshot: /assets/img/projects/network1.png
caption: We use single-cell and bulk-cell RNA sequencing to study gene expression in tissues and cells during infection.
description: >
  One of the original [EnrichmentMap](https://www.baderlab.org/Software/EnrichmentMap) analyses we used to find pathways that were unique and shared between inflamed skin from patients with cutaneous leishmaniasis, versus psoriasis patients (See paper [here](https://doi.org/10.1038/jid.2014.305))
# accent_color: '#4fb1ba'
# accent_image:
#   background: 'linear-gradient(to bottom,#193747 0%,#233e4c 30%,#3c929e 50%,#d5d5d4 70%,#cdccc8 100%)'
#   overlay:    true
featured: true
---

We carry out global transcriptional profiling of cells and tissues to identify the full complement of genes involked in response to intracellular infection or to distinct inflammatory signals (cytokines).  Bioinformatics and statistical tools are used to mine this data to identify key regulators of host immunity. Our efforts in this area have led to the identification of key pathways involved in pathogenesis during infection with *Leishmania*, *Toxoplasma* and *Trichinella*.  

## Dual RNAseq in human cutaneous leishmaniasis
We have a long-standing collaboration with the leishmania researchers Phil Scott (PennVet) and Edgar Carvalho (Brazil).  Our goal in this collaboration has been to apply high-throughput sequencing to identify genes and pathways involved in skin pathology during cutaneous leishmaniasis.  Our work in this area led to the discovery the inflammasome as a key driver of disease, and recently identifed biomarkers associated with treatment outcome in human disease.  As we move forward with this work, we are looking forward to integrating diverse 'omic datasets and using single cell methods to understand the factors that contribute to the highly variable clinical outcomes observed in patients affected by this neglected tropical disease.

## from RNAseq to biomarker diagnostics
Our identification of biomarkers of treatment outcome for cutaneous leishmaniasis has led us to become interested in how to effectively translate our basic research findings into assays that could feasibly be deployed as diagnostics in low-resource areas.  Toward this end, we have partnered with [Biomeme](https://biomeme.com/) to develop field-ready assays on their portable QPCR platform.  Currently we are using this platform to assay immune transcripts detectable on swabs of human skin.  Stay tuned for exciting results in this area!

## Transcriptomics of dysbiosis
We're interested in applying our expertise in RNAseq to profile both host responses and microbial community structure and function, all from the same sample of RNA.  We're only just beginning this work, but we hope to develop methods that will allow us gain critical insights into how changes in microbiome structure ('dysbiosis') affect gene expression at barrier surfaces such as the skin and gut. 

## Single cell RNAseq and ATACseq
In many of our projects, bulk RNAseq has yielded critical insight into pathobiology.  We are currently working to expand on this past work with single cell RNAseq and ATACseq using our newly acquired [10X chromium controller](https://www.10xgenomics.com/instruments/chromium-controller/) and [1CellBio InDrop](https://1cell-bio.com/) platforms.
