<!doctype html>
<!--[if lt IE 7]><html class="no-js lt-ie9 lt-ie8 lt-ie7" lang="en"> <![endif]-->
<!--[if (IE 7)&!(IEMobile)]><html class="no-js lt-ie9 lt-ie8" lang="en"><![endif]-->
<!--[if (IE 8)&!(IEMobile)]><html class="no-js lt-ie9" lang="en"><![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en"><!--<![endif]-->
<head>
<meta charset="utf-8">
<title>Cloud computing for metagenomics - 4hr workshop &#8211; Home</title>
<meta name="description" content="This workshop, led by Dan Beiting and Kyle Bittinger at UPenn on Nov 7th, uses cloud compute resources and a Snakemake-based pipeline called Sunbeam for analysis of shotgun metagenomic data">
<meta name="keywords" content="metagenomics">


<!-- Twitter Cards -->
<meta name="twitter:title" content="Cloud computing for metagenomics - 4hr workshop">
<meta name="twitter:description" content="This workshop, led by Dan Beiting and Kyle Bittinger at UPenn on Nov 7th, uses cloud compute resources and a Snakemake-based pipeline called Sunbeam for analysis of shotgun metagenomic data">
<meta name="twitter:site" content="@hostmicrobe">
<meta name="twitter:creator" content="@hostmicrobe">

<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://localhost:4000/images/default-thumb.png">

<!-- Open Graph -->
<meta property="og:locale" content="en_US">
<meta property="og:type" content="article">
<meta property="og:title" content="Cloud computing for metagenomics - 4hr workshop">
<meta property="og:description" content="This workshop, led by Dan Beiting and Kyle Bittinger at UPenn on Nov 7th, uses cloud compute resources and a Snakemake-based pipeline called Sunbeam for analysis of shotgun metagenomic data">
<meta property="og:url" content="http://localhost:4000/microbiome/cloudComputing_workshop/">
<meta property="og:site_name" content="Home">

<meta name="google-site-verification" content="x_a2aVtoLyANByc9VDBjyPm80IlGbNYUm4Vr1tSRhi0" />


<link rel="canonical" href="http://localhost:4000/microbiome/cloudComputing_workshop/">
<link href="http://localhost:4000/feed.xml" type="application/atom+xml" rel="alternate" title="Home Feed">

<!-- http://t.co/dKP3o1e -->
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="google-site-verification" content="x_a2aVtoLyANByc9VDBjyPm80IlGbNYUm4Vr1tSRhi0" />

<!-- For all browsers -->
<link rel="stylesheet" href="http://localhost:4000/assets/css/main.css">

<meta http-equiv="cleartype" content="on">

<!-- HTML5 Shiv and Media Query Support -->
<!--[if lt IE 9]>
	<script src="http://localhost:4000/assets/js/vendor/html5shiv.min.js"></script>
	<script src="http://localhost:4000/assets/js/vendor/respond.min.js"></script>
<![endif]-->

<!-- Modernizr -->
<script src="http://localhost:4000/assets/js/vendor/modernizr-2.7.1.custom.min.js"></script>

<link href='//fonts.googleapis.com/css?family=PT+Sans+Narrow:400,700%7CPT+Serif:400,700,400italic' rel='stylesheet' type='text/css'>

<!-- Icons -->
<!-- 16x16 -->
<link rel="shortcut icon" href="http://localhost:4000/favicon.ico">
<!-- 32x32 -->
<link rel="shortcut icon" href="http://localhost:4000/favicon.png">
<!-- 57x57 (precomposed) for iPhone 3GS, pre-2011 iPod Touch and older Android devices -->
<link rel="apple-touch-icon-precomposed" href="http://localhost:4000/images/apple-touch-icon-precomposed.png">
<!-- 72x72 (precomposed) for 1st generation iPad, iPad 2 and iPad mini -->
<link rel="apple-touch-icon-precomposed" sizes="72x72" href="http://localhost:4000/images/apple-touch-icon-72x72-precomposed.png">
<!-- 114x114 (precomposed) for iPhone 4, 4S, 5 and post-2011 iPod Touch -->
<link rel="apple-touch-icon-precomposed" sizes="114x114" href="http://localhost:4000/images/apple-touch-icon-114x114-precomposed.png">
<!-- 144x144 (precomposed) for iPad 3rd and 4th generation -->
<link rel="apple-touch-icon-precomposed" sizes="144x144" href="http://localhost:4000/images/apple-touch-icon-144x144-precomposed.png">

</head>

<body class="post">

<!--[if lt IE 9]><div class="browser-upgrade alert alert-info">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</div><![endif]-->

<a name="top"></a>
<div class="navigation-wrapper">
	<div class="site-name">
		<a href="http://localhost:4000/">Home</a>
	</div><!-- /.site-name -->
	<div class="top-navigation">
		<nav role="navigation" id="site-nav" class="nav">
		    <ul>
		        
				    
				    <li><a href="http://localhost:4000/research/"  >Research</a></li>
				
				    
				    <li><a href="http://localhost:4000/publications/"  >Publications</a></li>
				
				    
				    <li><a href="http://localhost:4000/people/"  >People</a></li>
				
				    
				    <li><a href="http://diytranscriptomics.com/"  target="_blank">Teaching</a></li>
				
				    
				    <li><a href="https://chmi-sops.github.io/"  target="_blank">Protocols</a></li>
				
		    </ul>
		</nav>
	</div><!-- /.top-navigation -->
</div><!-- /.navigation-wrapper -->




<div id="main" role="main">
  <div class="article-author-side">
    <div>
  <p><a href="http://www.vet.upenn.edu"><img src="http://localhost:4000/images/PennVet_logo.png" style="border: none;" width="200"></a>
  <p><a href="http://www.microbiomedb.org/"><img src="http://localhost:4000/images/MicrobiomeDB_logo.png" style="border: none;" width="200"></a>
  <p><a href="http://www.diytranscriptomics.com/"><img src="http://localhost:4000/images/DIYtrans_logo.png" style="border: none;" width="200"></a>
  <p><a href="http://chmi-sops.github.io/"><img src="http://localhost:4000/images/SOP_logo.png" style="border: none;" width="200"></a>
</div>

  </div>
  <article class="post">
    <div class="headline-wrap">
      
        <h1 ><a href="http://localhost:4000/microbiome/cloudComputing_workshop/" rel="bookmark" title="Cloud computing for metagenomics - 4hr workshop">Cloud computing for metagenomics - 4hr workshop</a></h1>
      
    </div><!--/ .headline-wrap -->
    <div class="article-wrap">

    
        

    
        
    
        
    
        
    
        
    
        
    


    

    <div id="articlecontent"><section id="table-of-contents" class="toc">
  <header>
    <h3><i class="fa fa-book"></i> Table of Contents</h3>
  </header>
<div id="drawer">
<ul id="markdown-toc">
  <li><a href="#before-starting" id="markdown-toc-before-starting">Before starting</a></li>
  <li><a href="#set-up-cloud-computer" id="markdown-toc-set-up-cloud-computer">Set-up cloud computer</a></li>
  <li><a href="#install-sunbeam" id="markdown-toc-install-sunbeam">Install Sunbeam</a></li>
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#initialize-project" id="markdown-toc-initialize-project">Initialize project</a></li>
  <li><a href="#download-reference-data" id="markdown-toc-download-reference-data">Download reference data</a></li>
  <li><a href="#run-the-pipeline" id="markdown-toc-run-the-pipeline">Run the pipeline</a></li>
  <li><a href="#generate-a-report" id="markdown-toc-generate-a-report">Generate a report</a></li>
</ul>

  </div>
</section>
<!-- /#table-of-contents -->

<h2 id="before-starting">Before starting</h2>
<p>In this workshop we’ll use the Google Cloud to analyze raw metagenomic sequence data to identify the microbial composition of stool from heathly humans compared to Crohn’s disease patients.  In addition to generating this microbial census, you’ll also assemble sequences into contigs, which can then be used to infer functional potential.  To accomplish these tasks we’ll use <a href="https://www.biorxiv.org/content/early/2018/05/18/326363">Sunbeam</a>, a snake-make based metagenomics pipeline developed by <a href="https://microbiome.research.chop.edu/our-team/kyle-bittinger.html">Kyle Bittinger</a> and his group at the PennCHOP Microbiome Center.</p>

<p>The material below is intended to walk you through this workshop, and provide a general web-based lesson plan for how one might conduct such a workshop.</p>

<p>To participate in this workshop, you’ll only need a few things:</p>
<ul>
  <li>a laptop computer</li>
  <li>an internet connection</li>
  <li>a google account (free)</li>
  <li><a href="https://cloud.google.com">a google cloud account</a> (free sign-up comes with a $300 credit!)</li>
</ul>

<h2 id="set-up-cloud-computer">Set-up cloud computer</h2>

<p>We’ll begin the workshop with a demonstration of how to launch your first Google Cloud instance to build your cloud computer to have the following specs:</p>

<ul>
  <li>8 cores</li>
  <li>50Gb of RAM</li>
  <li>100Gb solid state harddrive</li>
  <li>Running Linux 18.04 LTS operating system</li>
</ul>

<p>Once you have finalized this instance, you have effectively rented a computer from Google, and we are all using exactly the same type of computer with the same operating system and compute resources.  In the case of the computer we set-up above, you will be charged 36 cents per hour, or about $260/month.  The more powerful the computer, the more you will be charged in rent, regardless of whether or not you actually use these resources.  <strong>If you fail to delete your instance after the workshop, your credit card will be charged after ~1 month (when the $300 credit will have been spent).</strong></p>

<h2 id="install-sunbeam">Install Sunbeam</h2>

<ul>
  <li>
    <p>Connect to your cloud computer using the ‘ssh’ button next to the instance.</p>
  </li>
  <li>
    <p><a href="https://github.com/eclarke/sunbeam">Download Sunbeam</a> from github using the code below</p>
  </li>
</ul>

<pre><code class="language-{bash}">cd ~
git clone -b stable https://github.com/sunbeam-labs/sunbeam sunbeam-stable
ls
</code></pre>
<ul>
  <li>Install Sunbeam</li>
</ul>

<pre><code class="language-{bash}">cd sunbeam-stable
bash install.sh
</code></pre>
<ul>
  <li>
    <p>In order to start using Sunbeam, we need to close our ssh terminal and reopen it.</p>
  </li>
  <li>
    <p>Since Sunbeam was installed as a Conda environment, we have to enter this environment to start using the software</p>
  </li>
</ul>

<pre><code class="language-{bash}">source activate sunbeam
</code></pre>
<p>This is a command you’ll want to remember for future sessions.  Each time you log into your cloud instance, you’ll need to activate the pipeline with <code class="language-plaintext highlighter-rouge">source activate sunbeam</code>.  Upon activation, you should see that your command prompt begins with “(sunbeam)”.  Anytime you want to exit out of sunbeam, simply type <code class="language-plaintext highlighter-rouge">source deactivate sunbeam</code> and hit return.</p>

<h2 id="get-data">Get data</h2>

<ul>
  <li>let’s install some additional software in our environment. SRA tools will allow us to easily retrieve raw data from NCBI’s <a href="https://www.ncbi.nlm.nih.gov/sra">Sequence Read Archive</a></li>
</ul>

<pre><code class="language-{bash}">conda install -c bioconda sra-tools
</code></pre>
<ul>
  <li>For this workshop, we’ll use data from a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4633303/">recent metagenomics study</a> in Crohn’s disease.  This was a large study, but for the purpose of the workshop we’ll only fetch data from 7 patients.  Note: contaminating human reads have already been removed from these files.  Let’s download these data to our cloud computer using the <code class="language-plaintext highlighter-rouge">fasterq-dump</code> function from the SRA tools software.</li>
</ul>

<pre><code class="language-{bash}">cd ~
mkdir workshop-data
cd workshop-data
fasterq-dump SRR2145310
fasterq-dump SRR2145329
fasterq-dump SRR2145381
fasterq-dump SRR2145353
fasterq-dump SRR2145354
fasterq-dump SRR2145492
fasterq-dump SRR2145498
</code></pre>

<h2 id="initialize-project">Initialize project</h2>

<pre><code class="language-{bash}">cd ~
mkdir workshop-project
sunbeam init workshop-project --data_fp workshop-data
</code></pre>

<ul>
  <li>Use your nano text editor to explore the samples file and configuration file.</li>
</ul>

<h2 id="download-reference-data">Download reference data</h2>

<ul>
  <li>
    <p>We need two reference databases to run our analysis: a database of host DNA
sequence to remove, and a database of bacterial DNA to match against.</p>
  </li>
  <li>
    <p>We’ll get the human genome data from UCSC.  Filtering against the entire human
genome takes too long, so we’ll only filter against chromosome 1.</p>
  </li>
</ul>

<pre><code class="language-{bash}">cd ~
mkdir human
cd human
wget http://hgdownload.cse.ucsc.edu/goldenPath/hg38/chromosomes/chr1.fa.gz
gunzip chr1.fa.gz
</code></pre>

<ul>
  <li>Sunbeam requires that the host DNA sequence files end in “.fasta”, so it can
find them automatically.  Let’s use the <code class="language-plaintext highlighter-rouge">mv</code> command to rename this file.</li>
</ul>

<pre><code class="language-{bash}">mv chr1.fa chr1.fasta
</code></pre>

<ul>
  <li>The database of bacterial genomes comes pre-built from the homepage of our
taxonomic assignment software, Kraken.  We’ll download that using the <code class="language-plaintext highlighter-rouge">wget</code> program.</li>
</ul>

<pre><code class="language-{bash}">cd ~
wget https://ccb.jhu.edu/software/kraken/dl/minikraken_20171101_4GB_dustmasked.tgz
tar xvzf minikraken_20171101_4GB_dustmasked.tgz
</code></pre>

<ul>
  <li>Now that we have reference databases, we need to add them to our configuration
file.  We’ll use <code class="language-plaintext highlighter-rouge">nano</code> to open and modify this file directly in our termial</li>
</ul>

<pre><code class="language-{bash}">cd ~
nano workshop-project/sunbeam_config.yml
</code></pre>

<ul>
  <li>The configuration values are below.  You’ll need to navigate to the right spot
in your configuration file, and substitute the directory name “kylebittinger”
with your google username.</li>
</ul>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>host_fp: "/home/kylebittinger/human"
</code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>kraken_db_fp: "/home/kylebittinger/minikraken_20171101_4GB_dustmasked"
</code></pre></div></div>

<h2 id="run-the-pipeline">Run the pipeline</h2>

<ul>
  <li>We are ready to actually run the pipeline.  All the information about how
to run the pipeline is in our configuration file, so we’ll provide that to
Sunbeam (<code class="language-plaintext highlighter-rouge">--configfile</code> argument).  We’ll also let Sunbeam know how many CPU
cores we’d like to use (<code class="language-plaintext highlighter-rouge">--jobs</code> argument).</li>
</ul>

<pre><code class="language-{bash}">cd ~
sunbeam run --configfile workshop-project/sunbeam_config.yml --jobs 7
</code></pre>

<h2 id="generate-a-report">Generate a report</h2>

<p>To look at some of our results, we’ll install a Sunbeam extension and generate
a report.</p>

<pre><code class="language-{bash}">cd ~/sunbeam-stable/extensions
git clone https://github.com/sunbeam-labs/sbx_report
conda install --file sbx_report/requirements.txt
</code></pre>

<p>This is going to install the R programming language on our remote computer,
which will take a bit of time.  When the installation is complete,</p>

<pre><code class="language-{bash}">cd ~
sunbeam run --configfile workshop-project/sunbeam_config.yml final_report
</code></pre>
</div> 
      <hr style="margin-top: 100px;"/>
      <footer role="contentinfo">
        <div class="social-share">
  <h4>Share on</h4>
  <ul>
    <li>
      <a href="https://twitter.com/intent/tweet?text=http://localhost:4000/microbiome/cloudComputing_workshop/" class="twitter" title="Share on Twitter"><i class="fa fa-twitter"></i><span> Twitter</span></a>
    </li>
    <li>
      <a href="https://www.facebook.com/sharer/sharer.php?u=http://localhost:4000/microbiome/cloudComputing_workshop/" class="facebook" title="Share on Facebook"><i class="fa fa-facebook"></i><span> Facebook</span></a>
    </li>
    <li>
      <a href="https://plus.google.com/share?url=http://localhost:4000/microbiome/cloudComputing_workshop/" class="google-plus" title="Share on Google Plus"><i class="fa fa-google-plus"></i><span> Google+</span></a>
    </li>
  </ul>
</div><!-- /.social-share -->
        <p class="byline"><strong>Cloud computing for metagenomics - 4hr workshop</strong> was published on <time datetime="2017-11-06T00:00:00-05:00">November 06, 2017</time> and last modified on <time datetime="2018-11-05">November 05, 2018</time>.</p>
      </footer>
    </div><!-- /.article-wrap -->
  
    <section id="disqus_thread"></section><!-- /#disqus_thread -->
  
  </article>
</div><!-- /#main -->

<div class="footer-wrap">
  <div class="related-articles">
  <h4>You might also enjoy <small class="pull-right">(<a href="http://localhost:4000/posts/">View all posts</a>)</small></h4>
    <ul>
    
      <li><a href="http://localhost:4000/microbiome/cloudComputing_part2/" title="Cloud computing for metagenomics - Part II">Cloud computing for metagenomics - Part II</a></li>
    
      <li><a href="http://localhost:4000/microbiome/cloudComputing_part1/" title="Cloud computing for metagenomics - Part I">Cloud computing for metagenomics - Part I</a></li>
    
    </ul>
    <hr />
  </div><!-- /.related-articles -->
  <footer>
    

<span>&copy; 2020 Beiting. Powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a>. The theme is derived from <a href="http://mademistakes.com/minimal-mistakes/" rel="nofollow">Minimal Mistakes</a>.</span>

  </footer>
</div><!-- /.footer-wrap -->

<script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script>window.jQuery || document.write('<script src="http://localhost:4000/assets/js/vendor/jquery-1.9.1.min.js"><\/script>')</script>
<script src="http://localhost:4000/assets/js/scripts.min.js"></script>

  
<script type="text/javascript">
    /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
    var disqus_shortname = 'http-hostmicrobe-org'; // required: replace example with your forum shortname

    /* * * DON'T EDIT BELOW THIS LINE * * */
    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();

    /* * * DON'T EDIT BELOW THIS LINE * * */
    (function () {
        var s = document.createElement('script'); s.async = true;
        s.type = 'text/javascript';
        s.src = '//' + disqus_shortname + '.disqus.com/count.js';
        (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
    }());
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>





<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-420484-15', 'auto');
  ga('require', 'linkid', 'linkid.js');
  ga('send', 'pageview');
</script>

	


</body>
</html>
